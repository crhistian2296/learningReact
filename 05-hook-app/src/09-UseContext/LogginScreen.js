const LogginScreen = () => {
  return (
    <div>
      <h1>LogginScreen</h1>
      <hr />
    </div>
  );
};

export default LogginScreen;
