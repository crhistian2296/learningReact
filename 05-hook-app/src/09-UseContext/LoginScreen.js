import { useContext } from 'react';
import { UserContext } from './UserContext';

const LoginScreen = () => {
  const { user, setUser } = useContext(UserContext);
  // console.log(user);
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button
        className='btn btn-primary'
        onClick={() => {
          setUser({ id: 1212, name: 'Paco' });
        }}
      >
        Login
      </button>
      <button
        className='btn btn-primary mx-2'
        onClick={() => {
          setUser({});
        }}
      >
        Logut
      </button>
      <pre>{JSON.stringify(user, null, 3)}</pre>
    </div>
  );
};

export default LoginScreen;
