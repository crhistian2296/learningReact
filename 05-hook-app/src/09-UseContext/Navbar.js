import { NavLink } from 'react-router-dom';

const Navbar = () => {
  return (
    //* Navbar cogida de getBootstrap
    <nav className='navbar navbar-expand navbar-dark bg-dark'>
      <div className='container-fluid'>
        <NavLink className='navbar-brand' to='/'>
          useContext
        </NavLink>
        <div className='collapse navbar-collapse' id='navbarNav'>
          <ul className='navbar-nav'>
            <NavLink className='nav-item nav-link ' to='/'>
              Home
            </NavLink>
            <NavLink className='nav-item nav-link ' to='/about'>
              About
            </NavLink>
            <NavLink className='nav-item nav-link ' to='/login'>
              Login
            </NavLink>
          </ul>
        </div>
      </div>
    </nav>
  );
};

export default Navbar;
