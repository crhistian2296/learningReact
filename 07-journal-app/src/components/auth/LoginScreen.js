import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { startGoogleLogin, startLoginEmailPassword } from '../../actions/auth';
import useForm from '../../hooks/useFrom';

const LoginScreen = () => {
  const { formValues, handleInputChange } = useForm({
    email: 'cristof@gmail.com',
    password: '1234',
  });

  const { email, password } = formValues;

  const dispatch = useDispatch();

  const handleLogin = (e) => {
    e.preventDefault();
    dispatch(startLoginEmailPassword(123456, 'cristof'));
  };

  const handleGoogleLogin = () => {
    dispatch(startGoogleLogin());
  };

  return (
    <>
      <h3 className='box-container__title'>Login</h3>

      <form onSubmit={handleLogin}>
        <input
          type='text'
          placeholder='Email'
          name='email'
          className='box-container__input'
          value={email}
          onChange={handleInputChange}
          autoComplete='off'
        />
        <input
          type='password'
          placeholder='Password'
          name='password'
          className='box-container__input'
          value={password}
          onChange={handleInputChange}
        />
        <button type='submit' className='btn btn-primary btn-block mt-1'>
          Login
        </button>
        <div className='box-container__social-networks'>
          <p>Login with social networks</p>

          <div className='google-btn' onClick={handleGoogleLogin}>
            <div className='google-icon-wrapper'>
              <img
                className='google-icon'
                src='https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg'
                alt='google button'
              />
            </div>
            <p className='google-btn-text'>
              <b>Sign in with google</b>
            </p>
          </div>
        </div>
        <Link to='/auth/register' className='link'>
          Create new account
        </Link>
      </form>
    </>
  );
};

export default LoginScreen;
