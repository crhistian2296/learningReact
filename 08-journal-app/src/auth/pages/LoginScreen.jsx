import { Google } from '@mui/icons-material';
import {
  Alert,
  Button,
  Grid,
  Link,
  TextField,
  Typography,
} from '@mui/material';
import { useMemo } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link as RouteLink } from 'react-router-dom';
import { useForm } from '../../hooks';
import {
  startGoogleSignIn,
  startLoginWithEmailAndPassword,
} from '../../store/auth';
import AuthLayout from '../layout/AuthLayout';

const formData = {
  email: 'user@gmail.com',
  password: '123456',
};

const Login = () => {
  const { email, password, onInputChange } = useForm(formData);

  const { status, errorMessage = null } = useSelector(state => state.auth);
  const dispatch = useDispatch();

  const isAuthenticating = useMemo(() => status === 'checking', [status]);

  const onSubmit = event => {
    event.preventDefault();
    if (!email || !password) return console.warn('Incomplete credentials');

    console.debug('Credentials', { email, password });
    return dispatch(startLoginWithEmailAndPassword({ email, password }));
  };

  const onGoogleSignIn = async () => {
    console.debug('Google sign in');
    return dispatch(startGoogleSignIn());
  };

  return (
    <AuthLayout title='Login'>
      <form
        aria-label='submit-form'
        onSubmit={onSubmit}
        className='animate__animated animate__fadeIn animate__faster'
      >
        <Grid container>
          <Grid item xs={12} mt={2}>
            <TextField
              label='mail'
              type='email'
              fullWidth
              autoComplete='email'
              name='email'
              value={email}
              onChange={onInputChange}
            />
          </Grid>
          <Grid item xs={12} mt={2}>
            <TextField
              label='password'
              type='password'
              fullWidth
              autoComplete='current-password'
              name='password'
              inputProps={{ 'data-testid': 'password' }}
              value={password}
              onChange={onInputChange}
            />
          </Grid>
        </Grid>
        <Grid container spacing={2} mt={1}>
          <Grid
            item
            xs={12}
            sm={12}
            display={errorMessage !== null ? '' : 'none'}
          >
            <Alert severity='error'>{errorMessage}</Alert>
          </Grid>
          <Grid item xs={12} sm={6}>
            <Button
              type='submit'
              disabled={isAuthenticating}
              variant='contained'
              fullWidth
            >
              Login
            </Button>
          </Grid>
          <Grid item xs={12} sm={6}>
            <Button
              disabled={isAuthenticating}
              onClick={onGoogleSignIn}
              aria-label='google-btn'
              variant='contained'
              fullWidth
            >
              <Google />
              <Typography marginLeft={1} variant='body2'>
                Google
              </Typography>
            </Button>
          </Grid>
          <Grid container direction='row' justifyContent='end'>
            <Link
              component={RouteLink}
              color='inherit'
              mt={1}
              to='/auth/register'
            >
              Create account
            </Link>
          </Grid>
        </Grid>
      </form>
    </AuthLayout>
  );
};

export default Login;
