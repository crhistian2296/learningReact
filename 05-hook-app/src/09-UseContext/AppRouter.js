import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import AboutScreen from './AboutScreen';
import HomeScreen from './HomeScreen';
import LogginScreen from './LogginScreen';

const AppRouter = () => (
  <>
    <Router>
      <Routes>
        <Route path='/about' element={<AboutScreen />} />
        <Route path='/loggin' element={<LogginScreen />} />
        <Route path='/' element={<HomeScreen />} />
      </Routes>
    </Router>
  </>
);
export default AppRouter;
