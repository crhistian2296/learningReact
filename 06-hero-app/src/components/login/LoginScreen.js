import { useContext } from 'react';
import { useNavigate } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import { login } from '../../types/type';

const LoginScreen = () => {
  // react-router hook
  const navigate = useNavigate();
  const { dispatch } = useContext(AuthContext);

  const loginAction = {
    type: login,
    payload: {
      name: 'crhistian2296',
    },
  };

  const handleLogin = () => {
    dispatch(loginAction);

    const lastPath = localStorage.getItem('lastPath') ?? '/';

    navigate(lastPath, { replace: true });
  };

  return (
    <div className='m-5'>
      <h1>LoginScreen</h1>
      <hr />
      <button className='btn btn-primary' onClick={handleLogin}>
        Login
      </button>
    </div>
  );
};

export default LoginScreen;
