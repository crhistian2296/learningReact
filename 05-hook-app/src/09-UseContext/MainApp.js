import AppRouter from './AppRouter';

export const MainApp = () => {
  return (
    <main>
      <AppRouter />
    </main>
  );
};
