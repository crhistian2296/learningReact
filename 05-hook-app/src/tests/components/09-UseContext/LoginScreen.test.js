import { mount } from 'enzyme';
import LoginScreen from '../../../09-UseContext/LoginScreen';
import { UserContext } from '../../../09-UseContext/UserContext';

describe('Pruebas en <LoginScreen/>', () => {
  const user = {
    name: 'Paco',
    email: 'paco23@gmail.com',
  };

  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider
      value={{
        user,
        setUser,
      }}
    >
      <LoginScreen />
    </UserContext.Provider>
  );
  test('Debe de renderizar correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe ejecutar setUser con el argumento especificado', () => {
    // wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(0).prop('onClick')();

    expect(setUser).toHaveBeenCalledWith({ id: 1212, name: 'Paco' });
  });
});
