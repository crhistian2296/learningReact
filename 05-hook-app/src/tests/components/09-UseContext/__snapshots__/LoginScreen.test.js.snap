// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen/> Debe de renderizar correctamente 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
  <button
    className="btn btn-primary mx-2"
    onClick={[Function]}
  >
    Logut
  </button>
  <pre>
    {
   "name": "Paco",
   "email": "paco23@gmail.com"
}
  </pre>
</div>
`;
